src/vs/workbench/browser/parts/editor/baseEditor.ts

```diff
@@ -83,6 +84,13 @@ export abstract class BaseEditor extends Composite implements IEditorPane {
 	 */
 	protected abstract createEditor(parent: HTMLElement): void;
 
+	/**
+	 * Should be overridden by editors that have their own ScopedContextKeyService
+	 */
+	invokeWithinContext<T>(fn: (accessor: ServicesAccessor) => T): T | null {
```

Comment:
I think this is dangerous because the calling side will assume that when `null` is returned, the editor did not invoke in the context, but it could well be that a function that was passed in to invoke returns `null`. So a better approach imho is to pass in a function that is being called as fallback if no implementation exists? 

Alternatively we could make this method optional `?` and the caller has to probe if this method exists.

Reply:
Yeah, I was wondering whether that is an issue since "no implementation" is basically the same as "implementation but no service", but it's better to be explicit. I like making the method optional.